Providing data, support for entries and refresh, test coverage

stulees opened the pull request:
- component
	- component now fires "rise-rss-response" passing a full javascript object representation of rss feed
	- component now fires "rise-rss-error" if there was an error returned using `gadgets.io.makeRequest`
	- support added for `entries`, data returned will only include the amount of feed entries specified
	- support added for a `refresh`, "rise-rss-response" will be fired periodically with new request data based on `refresh` value in minutes
- test coverage
	- string based XML files added for both an RSS 2.0 feed and an Atom feed
	- javascript object files added for both an RSS 2.0 feed and an Atom feed, which match their corresponding XML files
	- mock added for `gadgets.io.makeRequest` function
	- unit and integration tests added, full coverage

stulees commented:
@donnapep Please review. Note, I actually required a fair amount of unit testing after all. However, I've maintained the ability for private state in case I do need private vars down the road. To provide some indication of functions that are _intended_ to be private, I've prefixed their names with `_`. Thanks

donnapep commented:
@stulees I agree that this is where it makes sense to use _ to indicate private members. One of the Polymer guys also suggested another approach that I'm not sure I followed. He's supposed to send me a gist as an example, or I'll try to find one in the docs. Will let you know.

Seems an odd place for the `var xml;` declaration in `test/rise-rss-integration.html`. Does it not work when put inside the first test suite with the other declarations?

Also, I would recommend adding a message as the last parameter in your assertions where you have more than one in your listeners so that it's easier to see which one fails.

There is also a `debugger` statement in your integration test.

Great tests otherwise.

stulees commented:
@donnapep 
- I've moved the `var xml;` global declaration to `makeRequest.js` file mock since the `xml` var is only used and associated with the `makeRequest` mock for integration tests. This should serve as a better place for it. 
- I've added messages as the last parameter in multiple assertions that are inside listener handlers. 
- I removed the debugger statement (left in by mistake)

Can you review again, thanks!

donnapep commented:
:+1:
